src/test/java/org/apache/commons/collections4/bloomfilter/DefaultIndexProducerTest.java

```diff
@@ -47,8 +52,7 @@ public boolean forEachIndex(IntPredicate predicate) {
 
     @Override
     protected int getBehaviour() {
-        // The default method streams a BitSet so is distinct and ordered.
```

Comment:
Note: This test has removed some testing of the default methods in IndexProducer.

It should be named `IndexProducerFromIndexArrayTest`. It is specifically targeting the default implementation of the IndexProducer when created from an index array. In that case the `asIndexArray()` method returns a clone of the original indices used to create it.

A `DefaultIndexProducerTest` would test the default implementation of `asIndexArray()` and look like this.

```Java
public class DefaultIndexProducerTest extends AbstractIndexProducerTest {

    /** Make forEachIndex unordered and contain duplicates. */
    private int[] values = {10, 1, 10, 1};

    @Override
    protected int[] getExpectedIndices() {
        return values;
    }

    @Override
    protected IndexProducer createProducer() {
        return new IndexProducer() {
            @Override
            public boolean forEachIndex(IntPredicate predicate) {
                Objects.requireNonNull(predicate);
                for (int i : values) {
                    if (!predicate.test(i)) {
                        return false;
                    }
                }
                return true;
            }
        };
    }

    @Override
    protected IndexProducer createEmptyProducer() {
        return new IndexProducer() {
            @Override
            public boolean forEachIndex(IntPredicate predicate) {
                Objects.requireNonNull(predicate);
                return true;
            }
        };
    }

    @Override
    protected int getBehaviour() {
        // The default method streams a BitSet so is distinct and ordered.
        // However the forEachIndex may not be distinct and ordered and
        // the test cannot distinguish the two cases.
        return 0;
    }
}
```

Note that this highlights a problem with consolidating all the behaviour flags. Currently the test suite cannot be used to test the asIndexArray behaviour is different from the forEachIndex behaviour.

I think we should revert the consolidation of the flags. However to avoid the duplication of flag setting through the various test cases we can change the AbstractIndexProducerTest test to do this:
```Java
public abstract class AbstractIndexProducerTest {

    protected static final int ORDERED = 0x1;
    protected static final int DISTINCT = 0x2;

    protected abstract int getAsIndexArrayBehaviour();

    protected int getForEachIndexBehaviour() {
        return getAsIndexArrayBehaviour();
    }

}
```
And AbtractBitCountProducerTest:
```Java
public abstract class AbstractBitCountProducerTest extends AbstractIndexProducerTest {

    protected int getForEachCountBehaviour() {
        return getAsIndexArrayBehaviour();
    }

}
```

Then get the correct flag in the relevant test. It provides the ability to override each test behaviour individually but will default to the same behaviour if not explicitly changed.

Doing this change (DefaultIndexProducerTest and IndexProducerFromIndexArrayTest) should remove the requirement for the additional test `testFromIndexArray`. The method `testFromBitMapProducer` is already covered in IndexProducerFromBitMapProducerTest.